docs: Revise the LogQL Analyzer topic (#14374)
grafana/loki: docs/sources/query/analyzer.md

---
title: Simple LogQL simulator

> menuTitle: LogQL Simulator
> description: The LogQL Simulator is an online educational tool for experimenting with writing simple LogQL queries.

Reviewer: To match the casing in the title and first heading

```suggestion
menuTitle: LogQL simulator
description: The LogQL simulator is an online educational tool for experimenting with writing simple LogQL queries.
```

aliases:
- ../logql/analyzer/
weight: 200
---

<link rel="stylesheet" href="../analyzer/style.css">
<script src="../analyzer/handlebars.js"></script>

# Simple LogQL simulator

> The LogQL Simulator is an online tool that you can use to experiment with writing simple LogQL queries and seeing the results, without needing to run an instance of Loki.

Reviewer: ```suggestion
The LogQL simulator is an online tool that you can use to experiment with writing simple LogQL queries and seeing the results, without needing to run an instance of Loki.
```

> A set of example log lines are included for each of Loki's primary supported log parsers:

Reviewer: https://grafana.com/docs/writers-toolkit/review/lint-prose/rules/#grafanaproductpossessives

```suggestion
A set of example log lines are included for each of the primary log parsers supported by Loki:
```

<script src="../analyzer/script.js"> </script>

> ## Additional Sample Queries

Reviewer: ```suggestion
## Additional sample queries
```

These are some additional sample queries that you can use in the LogQL simulator.

### Logfmt

> ```logQL
> | logfmt | level = "debug"
> ```
> 
> Parses logfmt-formatted logs and returns only log lines where the "level" field is equal to "debug".
> 
> ```logQL
> | logfmt | msg="server listening on addresses"
> ```
> 
> Parses logfmt-formatted logs and returns only log lines with the message “server listening on address.”
> 
> ### JSON
> 
> ```logQL
> | json | level="INFO" | file="SpringApplication.java" | line_format `{{.class}}`
> ```
> 
> Parses JSON-formatted logs, filtering for lines where the 'level' field is "INFO" and the 'file field is "SpringApplication.java", then formats the line to return only the 'class' field.
> 
> ```logQL
> |~ `(T|t)omcat`
> ```
> 
> Performs a regular expression filter for the string 'tomcat' or 'Tomcat', without using a parser.
> 
> ### Unstructured text
> 
> ```logQL
> | pattern "<_> - <_> <_> \"<method> <url> <protocol>\" <status> <_> <_> \"<_>\" <_>" | method="GET"
> ```
> 
> Parses unstructured logs with the pattern parser, filtering for lines where the HTTP method is "GET".
> 
> ```logQL
> | pattern "<_> - <user> <_> \"<method> <url> <protocol>\" <status> <_> <_> \"<_>\" <_>" | user=~"kling.*"
> ```

Reviewer: info strings are usually expected to be lowercase

````suggestion
```logql
| logfmt | level = "debug"
```

Parses logfmt-formatted logs and returns only log lines where the "level" field is equal to "debug".

```logql
| logfmt | msg="server listening on addresses"
```

Parses logfmt-formatted logs and returns only log lines with the message “server listening on address.”

### JSON

```logql
| json | level="INFO" | file="SpringApplication.java" | line_format `{{.class}}`
```

Parses JSON-formatted logs, filtering for lines where the 'level' field is "INFO" and the 'file field is "SpringApplication.java", then formats the line to return only the 'class' field.

```logql
|~ `(T|t)omcat`
```

Performs a regular expression filter for the string 'tomcat' or 'Tomcat', without using a parser.

### Unstructured text

```logql
| pattern "<_> - <_> <_> \"<method> <url> <protocol>\" <status> <_> <_> \"<_>\" <_>" | method="GET"
```

Parses unstructured logs with the pattern parser, filtering for lines where the HTTP method is "GET".

```logql
| pattern "<_> - <user> <_> \"<method> <url> <protocol>\" <status> <_> <_> \"<_>\" <_>" | user=~"kling.*"
```
````
